File: app/dbmodels/Document.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'Document',
    'menu'          => 'Management'
);
$table['fields'] = array(
    'id'                     => DBModels::type('increments',
                                               array('visible' => false)),
    'entity_id'              => DBModels::type('integer'),
    'is_recursive'           => DBModels::type('boolean',
                                               array('visible' => false)),
    'is_deleted'             => DBModels::type('boolean',
                                               array('visible' => false)),
    'name'                   => DBModels::type('string'),
    'filename'               => DBModels::type('string'),
    'filepath'               => DBModels::type('string'),
    'comment'                => DBModels::type('text'),
    'document_category_id'    => DBModels::type('integer'),
    'mime'                   => DBModels::type('string'),
    'link'                   => DBModels::type('string'),
    'users_id'               => DBModels::type('integer'),
    'tickets_id'             => DBModels::type('integer'), /// TODO manage relationships
    'sha1sum'                => DBModels::type('string'),
    'is_blacklisted'         => DBModels::type('boolean'),
    'tag'                    => DBModels::type('string'),
);

$table['relationships'] = array(
    'document_category' => array(
        'type' => 'belongsTo',
        'item' => 'DocumentCategory'),
    'user' => array(
        'type' => 'belongsTo',
        'item' => 'User'),
    'entity' => array(
        'type'  => 'belongsTo',
        'item'  => 'Entity'),
);

File: fusionglpi/database/document.py
from fusionglpi import db

class Document(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    entity_id = db.Column(db.Integer, db.ForeignKey('entity.id'))
    is_recursive = db.Column(db.Boolean, nullable=False, server_default="0")
    name = db.Column(db.String(255, None, True))
    is_deleted = db.Column(db.Boolean, nullable=False, server_default="0")
    filename = db.Column(db.String(255, None, True))
    filepath = db.Column(db.String(255, None, True))
    comment = db.Column(db.Text)
    document_category_id = db.Column(db.Integer, db.ForeignKey('document_category.id'))
    mime = db.Column(db.String(255, None, True))
    link = db.Column(db.String(255, None, True))
    user_id = db.Column(db.Integer, db.ForeignKey('user.id'))
    #ticket_id = db.Column(db.Integer, db.ForeignKey('ticket.id'))
    sha1sum = db.Column(db.String(255, None, True))
    is_blacklisted = db.Column(db.Boolean, nullable=False, server_default="0")
    tag = db.Column(db.String(255, None, True))
